packages/block-library/src/details-summary/edit.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * WordPress dependencies
+ */
+import { RichText, useBlockProps } from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+
+function DetailsSummaryBlock( { attributes, setAttributes } ) {
+	const summary = attributes.summary ? attributes.summary : __( 'Details' );
+	return (
+		<summary
```

Comment:
The documentation for View says it is only for divs.
I could use tagName, if that is easier to read.

Reply:
nvm,  `View as`, found an example!

Reply:
It's not working. The <details> is not recognizing the `<summary>` so it outputs the default text.

Reply:
For some reason, `"summary"` is being rendered as an HTML attribute e.g. `as="summary"` on a `div`. My guess is there is a filter somewhere on what a valid `as` prop can be.

If we did still want to avoid the summary variable being the same as the tag name we could do something like the snippet below. I don't think this is a critical improvement though.

```diff
diff --git a/packages/block-library/src/details-summary/edit.js b/packages/block-library/src/details-summary/edit.js
index 9e091382ab..bbb0490021 100644
--- a/packages/block-library/src/details-summary/edit.js
+++ b/packages/block-library/src/details-summary/edit.js
@@ -4,10 +4,11 @@
 import { RichText, useBlockProps } from '@wordpress/block-editor';
 import { __ } from '@wordpress/i18n';
 
+const Summary = 'summary';
+
 function DetailsSummaryEdit( { attributes, setAttributes } ) {
-	const summary = attributes.summary ? attributes.summary : __( 'Details' );
 	return (
-		<summary
+		<Summary
 			{ ...useBlockProps() }
 			onClick={ ( event ) => event.preventDefault() }
 		>
@@ -15,12 +16,12 @@ function DetailsSummaryEdit( { attributes, setAttributes } ) {
 				aria-label={ __( 'Add summary' ) }
 				allowedFormats={ [] }
 				withoutInteractiveFormatting
-				value={ summary }
+				value={ attributes.summary || __( 'Details' ) }
 				onChange={ ( newSummary ) =>
 					setAttributes( { summary: newSummary } )
 				}
 			/>
-		</summary>
+		</Summary>
 	);
 }
 

```

Reply:
> "summary" is being rendered as an HTML attribute e.g. as="summary" on a div.

Yes I encountered that when both the edit save used View.

---
Personally I think using tagName would be clearer.

Reply:
Could the `<summary>` element be rendered by the `RichText`?

```jsx
return (
    <RichText
        tagName="summary"
        // ... 
    />
);
```

Reply:
I have not been able to make this work because when the RichText is the `<summary>`, you can't add `space` to the text. Space is used to open and close the details. In the editor, this behavior is prevented with  `preventDefault()`